File: ProDy/count.py
# Benchmark the counting of alanine residues in a PDB file

import time
from prody import *

pdb_filepath = "pdbs/1AKE.pdb"
struc = parsePDB(pdb_filepath)

def count():
    count = 0
    for res in struc.getHierView().iterResidues():
        if res.getResname() == "ALA":
            count += 1
    return count

start = time.time()
count()
elapsed = time.time() - start

print elapsed

File: MIToS/count.jl
# Benchmark the counting of alanine residues in a PDB file

using MIToS.PDB

pdb_filepath = "pdbs/1AKE.pdb"
struc = read(pdb_filepath, PDBFile)

counter() = count(res -> res.id.name == "ALA", struc)

# Run to JIT compile
counter()

elapsed = @elapsed counter()

println(elapsed)
